internal/p2p/peermanager.go

```diff
@@ -390,7 +400,12 @@ func (m *PeerManager) Add(address NodeAddress) (bool, error) {
 		return false, err
 	}
 	if address.NodeID == m.selfID {
-		return false, fmt.Errorf("can't add self (%v) to peer store", m.selfID)
+		m.logger.Error("not adding peer with self ID",
+			"peer", address.NodeID,
+			"self", m.selfID,
+			"address", address.String(),
+		)
```

Comment:
Do we really need to log an error when this could happen in a happy path?

Reply:
I think in the case where you accidentally don't get any peers, it'd be good to know that it's a configuration error? 

In the common case I think I agree with you, but I think the UX of "start a new node, have the self node as seed node," would be good to avoid.